examples/animatediff/ad/modules/diffusionmodules/motion_module.py

```diff
@@ -0,0 +1,523 @@
+        upcast_attention=False,
+        cross_frame_attention_mode=None,
+        temporal_position_encoding=False,
+        temporal_position_encoding_max_len=24,
+        dtype=ms.float32,
+    ):
+        super().__init__()
+
+        attention_blocks = []
+        norms = []
+
+        for block_name in attention_block_types:
+            attention_blocks.append(
+                VersatileAttention(
+                    attention_mode=block_name.split("_")[0],
+                    cross_attention_dim=cross_attention_dim if block_name.endswith("_Cross") else None,
+                    query_dim=dim,
+                    heads=num_attention_heads,
+                    dim_head=attention_head_dim,
+                    dropout=dropout,
+                    bias=attention_bias,
+                    upcast_attention=upcast_attention,
+                    cross_frame_attention_mode=cross_frame_attention_mode,
+                    temporal_position_encoding=temporal_position_encoding,
+                    temporal_position_encoding_max_len=temporal_position_encoding_max_len,
+                    dtype=dtype,
+                )
+            )
+            norms.append(LayerNorm32([dim], epsilon=1.0e-5))  # TODO: need fp32?
+
+        self.attention_blocks = nn.CellList(attention_blocks)
+        self.norms = nn.CellList(norms)
+
+        self.ff = FeedForward(dim, dropout=dropout, activation_fn=activation_fn)
+        # TODO: check correctness
+        self.ff_norm = LayerNorm32([dim], epsilon=1.0e-5)  # TODO: need fp32?
+
+    def construct(self, hidden_states, encoder_hidden_states=None, attention_mask=None, video_length=None):
+        for attention_block, norm in zip(self.attention_blocks, self.norms):
+            # print("D--: attention_block.is_cross_attention ", attention_block.is_cross_attention)
+            norm_hidden_states = norm(hidden_states)
+            hidden_states = (
+                attention_block(
+                    norm_hidden_states,
+                    encoder_hidden_states=encoder_hidden_states if attention_block.is_cross_attention else None,
+                    video_length=video_length,
+                )
+                + hidden_states
+            )
+
+        hidden_states = self.ff(self.ff_norm(hidden_states)) + hidden_states
+
+        output = hidden_states
+        return output
+
+
+class PositionalEncoding(ms.nn.Cell):
+    def __init__(self, d_model, dropout=0.0, max_len=24):
+        super().__init__()
+        self.dropout = ms.nn.Dropout(p=dropout)
+        position = np.expand_dims(np.arange(max_len), 1)
+
+        div_term = np.exp((np.arange(0, d_model, 2) * (-math.log(10000.0) / d_model)).astype(np.float32))
+        pe = np.zeros((1, max_len, d_model), dtype=np.float32)
+
+        pe[0, :, 0::2] = np.sin(position * div_term)
+        pe[0, :, 1::2] = np.cos(position * div_term)
+
+        self.pe = ms.Parameter(ms.Tensor(pe, dtype=ms.float32), name="pe", requires_grad=False)
```

Comment:
Maybe using `ms.Tensor` instead of `ms.Parameter` is OK here. I have discussed this with Rustam before. It seems that current ms version does not require the nn.Cell's tensor attribute  to be a `ms.Parameter`.

Reply:
https://github.com/mindspore-lab/mindone/pull/248#discussion_r1427481430

Reply:
It's a little risky. Not sure this change is ok for different MS versions on Ascend/GPU